wolfcrypt/src/port/Espressif/esp_crt_bundle/esp_crt_bundle.c

```diff
@@ -0,0 +1,1380 @@
+    }
+
+    if (store->current_cert == NULL) {
+        ESP_LOGE(TAG, "cert_manager_load store->current_cert is null");
+        return 0; /* preverify */
+    }
+
+    cm = store->store->cm;
+    peer = store->current_cert;
+    wolfSSL_X509_get_cert_items("peer", peer, &issuer, &subject);
+
+    /* It is the caller's responsibility to check conditions to add cert. */
+    if ((preverify == 0)  && (store->error == ASN_NO_SIGNER_E)) {
+        ESP_LOGCBI(TAG, "Confirmed call for ASN_NO_SIGNER_E");
+    }
+    else {
+        ESP_LOGW(TAG, "Warning: calling for non ASN_NO_SIGNER_E error.");
+    }
+
+    ESP_LOGCBI(TAG, "Cert %d:\n\tIssuer: %s\n\tSubject: %s\n",
+        store->error_depth,
+        issuer->name != NULL ? issuer->name : "[none]",
+        subject->name != NULL ? subject->name : "[none]");
+
+    /* Load the der cert to Certificate Manager:*/
+    ret = wolfSSL_CertManagerLoadCABuffer(cm, der, derSz,
+                                          WOLFSSL_FILETYPE_ASN1);
+    if (ret == WOLFSSL_SUCCESS) {
+        /* Attempt to validate the certificate again */
+        ret = wolfSSL_CertManagerVerifyBuffer(cm, der, derSz,
+            WOLFSSL_FILETYPE_ASN1);
+
+        if (ret == WOLFSSL_SUCCESS) {
+            ESP_LOGCBI(TAG, "Successfully validated cert: %s\n", subject->name);
+
+            /* If verification is successful then override error. */
+            preverify = 1;
+        }
+        else {
+            ESP_LOGE(TAG, "Failed to verify cert after loading new CA.");
+        }
+    }
+    else {
+        ESP_LOGE(TAG, "Failed to load CA");
+    }
+
+    /* We don't free the issue and subject, as they are
+     * pointers to current store->current_cert values. */
+    return preverify;
+}
+
+
+#if defined(WOLFSSL_DEBUG_CERT_BUNDLE) && defined(WOLFSSL_DEBUG_IGNORE_ASN_TIME)
+static CB_INLINE int wolfssl_ssl_conf_verify_cb_before_date(int preverify,
+                                                WOLFSSL_X509_STORE_CTX* store)
+{
+    if (store == NULL) {
+        ESP_LOGE(TAG, "wolfssl_ssl_conf_verify_cb_before_date store is null");
+        preverify = 0;
+    }
+    else if ((preverify == 0) && (store->error == ASN_BEFORE_DATE_E)) {
+        ESP_LOGW(TAG, "Overriding ASN_BEFORE_DATE_E!");
+        preverify = 1;
+    }
+
+    return preverify;
+}
+
+static CB_INLINE int wolfssl_ssl_conf_verify_cb_after_date(int preverify,
+                                                WOLFSSL_X509_STORE_CTX* store)
+{
+    if (store == NULL) {
+        ESP_LOGE(TAG, "wolfssl_ssl_conf_verify_cb_after_date store is null");
+        preverify = 0;
+    }
+    else if ((preverify == 0) && (store->error == ASN_AFTER_DATE_E)) {
+        ESP_LOGW(TAG, "Overriding ASN_AFTER_DATE_E!");
+        preverify = 1;
+    }
+
+    return preverify;
+}
+#endif /* WOLFSSL_DEBUG_CERT_BUNDLE && WOLFSSL_DEBUG_IGNORE_ASN_TIME */
+
+#ifdef CONFIG_WOLFSSL_DEBUG_CERT_BUNDLE
+void print_cert_subject_and_issuer(WOLFSSL_X509_STORE_CTX* store) {
```

Comment:
Brace new line. Move `int i` declaration to top of function

Reply:
done. Also added NULL check for `store`.